src/AppBundle/Repository/EventRepository.php

```diff
@@ -396,6 +396,83 @@ private function getUsernamesSql(Event $event): string
         return ltrim(implode(',', $usernames), ',');
     }
 
+    /**
+     * @param Event $event
+     * @return int[]
+     */
+    public function getBytesChanged(Event $event): array
+    {
+        $result = [];
+        foreach ($event->getWikis() as $wiki) {
+
+            /** @var EventWikiRepository $eventWikiRepo */
+            $eventWikiRepo = $this->em->getRepository('Model:EventWiki');
+            $eventWikiRepo->setContainer($this->container);
+
+            $dbName = $eventWikiRepo->getDbNameFromDomain($wiki->getDomain());
+
+            $pageIds = $wiki->getPages();
+
+            $result[$wiki->getDomain()] = $this->getBytesInternal($event, $dbName, $pageIds);
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param Event $event
+     * @param string $dbName
+     * @param int[] $pageIds
+     * @return int
+     */
+    private function getBytesInternal(Event $event, string $dbName, array $pageIds): int
+    {
+        $revisionTable = $this->getTableName('revision');
+        $pageTable = $this->getTableName('page');
+        $userIds = $event->getParticipantIds();
+        $usernames = array_column($this->getUsernamesFromIds($userIds), 'user_name');
+
+        $after = "SELECT IFNULL(rev_len,0)
+            FROM $dbName.$revisionTable
+            WHERE rev_page=page_id
+              AND rev_timestamp BETWEEN :start AND :end
+              AND rev_user_text IN (:usernames)
+            ORDER BY rev_timestamp DESC
+            LIMIT 1";
+
+        $before = "SELECT IFNULL(prev.rev_len, 0)
+            FROM $dbName.$revisionTable cur
+                JOIN $dbName.$revisionTable prev ON cur.rev_parent_id=prev.rev_id
+            WHERE cur.rev_page=page_id
+              AND cur.rev_timestamp BETWEEN :start AND :end
+              AND cur.rev_user_text IN (:usernames)
+            ORDER BY cur.rev_timestamp ASC
+            LIMIT 1";
+
+        $outerSql = "SELECT SUM(after) - SUM(before_)
+            FROM (
+                SELECT ($after) after, ($before) before_
+                    FROM $dbName.$pageTable
+                    WHERE page_id IN (:pageIds)
+                ) t1";
```

Comment:
Did you try using the page IDs in the `after` and `before` queries and then just do the calculation in the PHP code rather than in the SQL? I'm wondering if the query time is equivalent between the two approaches. I also wonder if that approach wouldn't make the code a bit simpler.

Reply:
You mean return all the revisions during the event?